### ISSUE TITLE
List and select available saves in Console

### ISSUE THREAD

**Issue by Extender92** [Extender92/Monopoly#17]
## Problem/Goal

Console cannot discover and choose among available saves through a reusable storage boundary. Add a safe list/select/load flow that never replaces the active match until a complete candidate load succeeds.

## Required behavior

- Console lists available saves through an injected save store rather than direct directory access.
- The user can select an entry or cancel explicitly.
- Missing, unreadable, invalid, incompatible, and general storage failures have distinct, clear frontend messages.
- A candidate match is fully loaded and validated before session replacement.
- Any failed or cancelled load leaves the active match unchanged.

## Scope

- Console save listing, selection, cancellation, and error presentation.
- Wiring to the injected persistence abstraction.
- Frontend-neutral save-catalog contracts and Infrastructure file enumeration required by the selection flow.
- Fake-store Console tests and session-replacement tests.

## Out of scope

- Serialization, atomic single-file persistence, and typed storage errors, owned by #33.
- Save schema and whole-match validation, owned by #52.
- Save naming and overwrite UX, owned by #16.
- Game-rule changes.

## Acceptance criteria

- Available saves are listed in a deterministic, documented order with sufficient identity for selection.
- Selection loads through the injected store and replaces the session only after complete validation.
- Cancel, missing entry, invalid content, incompatible version, and storage failure leave the active match unchanged.
- Console presents each failure without an unhandled exception or direct file-system access.
- Navigation returns through the shared explicit session loop.
- Fake-store tests cover empty, success, cancel, disappeared entry, invalid data, incompatible version, and storage failure.
- Full Release build and test suite pass without new warnings.

## Dependencies

- Blocked by shared session composition #18 and Infrastructure extraction #33.
- Final incompatible-version behavior follows Save Format Version 2 in #52.

## Related issues

- Related to #16
- Related to #20

**Comment by Extender92** [comment]
Deferred during neutral-Core backlog sanitation. This issue has not been implemented. It will not be copied to the clean repository and is not a cutover prerequisite. Reconsider the behavior only as a new, independently scoped neutral issue after cutover, based on the clean architecture and current product goals.
